File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task54_PERPLEXITY_llama-3-sonar-large-32k-chat.cpp
#include <iostream>
#include <ctime>
#include <cstdlib>

std::string generateOTP() {
    srand(static_cast<unsigned int>(time(0)));
    return std::to_string(rand() % 900000 + 100000);
}

std::string authenticate(std::string username, std::string password, std::string otp) {
    if (username == "admin" && password == "password" && otp == generateOTP()) {
        return "Authentication successful";
    } else {
        return "Authentication failed";
    }
}

int main() {
    std::string username, password, otp;
    std::cout << "Enter username: ";
    std::cin >> username;
    std::cout << "Enter password: ";
    std::cin >> password;
    std::cout << "Enter OTP: ";
    std::cin >> otp;
    std::cout << authenticate(username, password, otp) << std::endl;
    return 0;
}

File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task54_PERPLEXITY_llama-3-sonar-large-32k-chat.java
import java.util.Random;
import java.util.Scanner;

public class TwoFactorAuth {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Random random = new Random();

        System.out.println("Enter username:");
        String username = scanner.nextLine();
        System.out.println("Enter password:");
        String password = scanner.nextLine();

        if (username.equals("admin") && password.equals("password")) {
            int otp = random.nextInt(900000) + 100000;
            System.out.println("One Time Password (OTP) is: " + otp);
            System.out.println("Enter OTP:");
            int userOtp = scanner.nextInt();

            if (userOtp == otp) {
                System.out.println("Login successful!");
            } else {
                System.out.println("Invalid OTP. Login failed.");
            }
        } else {
            System.out.println("Invalid username or password. Login failed.");
        }
    }
}